### PR TITLE
add password ui

### PR DESCRIPTION
**Pull request by Entropia95**
Added password visualization

### CODE REVIEW

**Review by TheGrinch00 — APPROVED** [review]

**Review by TheGrinch00 — APPROVED** [review]

**Review by TheGrinch00 — APPROVED** [review]
Nice job

File: src/components/templates/protected-area/passwordVisualization.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from "react"
+import "./passwordVisualization.scss"
+
+//show password in different color based on matched char
+const PasswordVisualization = (props: any) => {
```
Comment (powext) [inline src/components/templates/protected-area/passwordVisualization.tsx]: We shouldn't definitely use any, an interface would be way better

File: src/components/templates/protected-area/passwordVisualization.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  let passw = props.valueActual
+  let str: any[] = []
```
Comment (powext) [inline src/components/templates/protected-area/passwordVisualization.tsx]: Here too

File: src/components/templates/protected-area/passwordVisualization.scss [hunk 3]
```diff
@@ -0,0 +1,15 @@
+.passw-alpha {
```
Comment (powext) [inline src/components/templates/protected-area/passwordVisualization.scss]: component's CSS must be wrapped to have styling incapsulation

File: src/components/templates/protected-area/passwordVisualization.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  let i: number
+  for (i = 0; i < passw.length; i++) {
+    if (passw[i].match(/[A-z]/)) {
+      str[i] = <span className="passw-alpha">{passw[i]}</span>
+    } else if (passw[i].match(/[0-9]/)) {
+      str[i] = <span className="passw-numbs">{passw[i]}</span>
+    } else {
+      str[i] = <span className="passw-special">{passw[i]}</span>
+    }
+  }
```
Comment (powext) [inline src/components/templates/protected-area/passwordVisualization.tsx]: At every render the child password is recreated, useMemo would fix the issue

**Review by powext — CHANGES_REQUESTED** [review]

**Review by powext — APPROVED** [review]

**Pull request merged**